### DISCUSSION TITLE
WiX v4-v5 XSD schemas

### DISCUSSION THREAD

**Discussion by chrpai** [wixtoolset/issues#7944] [WiX Development]
I noticed in v4 that with Votive being replaced by Heatwave that the XSDs  have been moved to FireGiant repos.  While this makes sense from a Heatwave installer perspective, it doesn't from a wix perspective.  For example, .NET DLLs included .XML docs in their nuget packages.  

The reason I ask it that it raises questions for me as a I watch the weekly meeting discussing the v5 roadmap.  I need to be able to update IsWiX to support v5 changes and generally I use these .XSD files as the contract to ensure that I'm parsing and emitting valid WiX code.  What's the plan here?  Will Heatwave rc's release on cadence with WiX?   Is the namespace changing again?

**Comment by barnson** [comment]
XSDs live here: https://github.com/wixtoolset/web/tree/master/src/xsd4

**Comment by chrpai** [comment]
Thanks Bob. I think I knew that somewhere deep inside.  Looking at https://github.com/wixtoolset/web/blob/master/src/xsd4/wix.xsd  I see schema changes with notes for v5 but the schema still says v4 in the uri.  Is that what I should plan for?

**Reply by barnson** [reply]
The namespace URI changes for breaking schema changes. So far, v5 hasn't introduced any.

**Comment by chrpai** [comment]
Breaking in which direction though.

Consider a designer that knows how to author v3 or v4 schema.  In v3 it authors the product element and in v4 it authors the package element. Along comes v5 and now there is a UpgradeStrategy attribute.   How does the designer know if it's dealing with v4 or v5 if the namespace hasn't changed?  It could 

1: assume v5 and read/write this new attribute
2: assume v4 and feign ignorance preserving any attribute that may exist
3: assume v5 only if it sees the attribute
4: other?

Of course ^^ logic would have to be implemented anywhere in the schema  a change has occurred.  This gets very messy fast.

Would a WiXToolSetVersion attribute somewhere help this?

**Reply by barnson** [reply]
So far, WiX v5 features are backward compatible, so WiX v5 can compile WiX v4 code without changes. That's the only guarantee.
